src/components/ProfileCard.tsx

```diff
@@ -0,0 +1,280 @@
+import React from 'react'
+import {
+  Avatar,
+  Box,
+  Button,
+  Card,
+  Container,
+  Grid,
+  IconButton,
+  Stack,
+  TextField,
+  Typography,
+  styled,
+} from '@mui/material'
+import EditIcon from '@mui/icons-material/Edit'
+import DoneIcon from '@mui/icons-material/Done'
+
+import CircleLoadAnimation from '../components/CircleLoadAnimation'
+
+import { LoginButton } from '../pages/LoginPage'
+import { ProfileData } from '../pages/ProfilePage'
+
+export const ProfileButton = styled(Button)({
+  padding: '0px 16px',
+  textTransform: 'none',
+  fontSize: 20,
+  fontFamily: ['Raleway', 'sans-serif'].join(','),
+  fontWeight: 500,
+})
+
+export const FlexBox = styled(Box)({
+  display: 'flex',
+  flexDirection: 'row',
+  flexWrap: 'wrap',
+  alignItems: 'center',
+  gap: '20px',
+})
+
+interface ProfileCardProps {
+  profileData: ProfileData
+  setProfileData: React.Dispatch<React.SetStateAction<ProfileData>>
+  oldProfileData: ProfileData
+
+  isEditing: boolean
+  toggleEdit: () => void
+  isWaiting: boolean
+
+  // We only need to pass in items that the API could reject.
+  // (it will never reject the user's first/last name)
+  usernameStatus: string
+  emailStatus: string
+}
+
+// Child Component for Profile Page, ID Card for viewing and editing metadata
+export default function ProfileCard({
+  profileData,
+  setProfileData,
+  oldProfileData,
+  isEditing,
+  toggleEdit,
+  isWaiting,
+  usernameStatus,
+  emailStatus,
+}: ProfileCardProps) {
+  const onTextFieldChange = (key, newString) => {
+    setProfileData((prevData: ProfileData) => {
+      return { ...prevData, [key]: newString }
+    })
+  }
+
+  const hasIssue = (str: string) => {
+    return str != 'Success' && str != 'Not Updated'
+  }
+
+  return (
+    <Card
+      sx={{
+        p: 4,
+        width: '32%',
+        position: 'relative',
+      }}
+    >
+      <IconButton
+        onClick={toggleEdit}
+        sx={{
+          position: 'absolute',
+          top: 15,
+          right: 15,
+          borderBottom: '1px solid #04364A',
+          borderRadius: 0,
+          padding: 0.5,
+          paddingLeft: 1,
+        }}
+      >
+        {isEditing ? (
+          <React.Fragment>
+            <Typography color={'#04364A'}>Done</Typography>
+            <Box width={'5px'}></Box>
+            {isWaiting ? (
+              <CircleLoadAnimation />
+            ) : (
+              <DoneIcon
+                style={{ fill: '#04364A', fontSize: '24px' }}
+              ></DoneIcon>
+            )}
+          </React.Fragment>
+        ) : (
+          <React.Fragment>
+            <Typography color={'#04364A'}>Edit</Typography>
+            <Box width={'5px'}></Box>
+            <EditIcon style={{ fill: '#04364A', fontSize: '24px' }}></EditIcon>
```

Comment:
Just a thought about the entire code base in general. I wonder if we can put the color hex codes in a useContext in App.tsx so we avoid hard coding them everywhere on the website. I have some of these in my components as well. We can talk more later.